### PR TITLE
T08 visual timetable UI

### PR DESCRIPTION
**Pull request by skellyhoe**
Sprint 1 is finally finished now :)

**Task description:**
Build the page where the admin would set a visual timetable:

- The admin will be presented with the items board
- Each element in the board is a clickable item
- When an image on the items board is clicked, it’s put on a list instead of being highlighted.
- When an image on the timetable is clicked, its popped off the list.
- The visual timetable consists of 5 pictures AT MOST, and is connected with arrows.

**Implemented all of that with the addition of:**

- Timetables can be saved for future use using a button that appears after 2 images are selected.
- A page that shows all saved timetables and allows the admin to delete unwanted timetables.

### CODE REVIEW

File: lib/pages/visual_timetable.dart [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    ImageDetails(name: "Orange", imageUrl: "https://images.unsplash.com/photo-1582979512210-99b6a53386f9?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=934&q=80"),
+    ImageDetails(name: "Footy", imageUrl: "https://upload.wikimedia.org/wikipedia/commons/a/ad/Football_in_Bloomington%2C_Indiana%2C_1996.jpg"),
+    ImageDetails(name: "Boxing", imageUrl: "https://e2.365dm.com/23/02/384x216/skysports-liam-wilson-emanuel-navarrete_6045983.jpg?20230204075325"),
+    ImageDetails(name: "Swimming", imageUrl: "https://cdn.britannica.com/83/126383-050-38B8BE25/Michael-Phelps-American-Milorad-Cavic-final-Serbia-2008.jpg"),
+    ImageDetails(name: "Fish and chips", imageUrl: "https://forkandtwist.com/wp-content/uploads/2021/04/IMG_0102-500x500.jpg"),
+  ];
+  ListOfListsOfImageDetails listOfTimetables = ListOfListsOfImageDetails(listOfLists: []);
+  // List<List<ImageDetails>> listOfTimetables = [];
```
Comment (Ali-Alkhars) [inline lib/pages/visual_timetable.dart]: Remove commented out code

File: lib/pages/visual_timetable.dart [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+          Expanded(
+            flex: 5,
+            child: Container(
+              width: 1100,
+              height: 200,
+              alignment: Alignment.center,
+              child: timetableList,
+            )
+          ),
+          // if (timetableList.imagesList.length >= 2) buildAddButton(timetableList),
```
Comment (Ali-Alkhars) [inline lib/pages/visual_timetable.dart]: remove here too

File: lib/widgets/image_square.dart [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  final ImageDetails image;
+  final double height;
+  final double width;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: width,
+      height: height,
+      // padding: const EdgeInsets.all(10),
```
Comment (Ali-Alkhars) [inline lib/widgets/image_square.dart]: here too, also line 30

File: lib/widgets/timetable_row.dart [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        color: Colors.teal,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      height: 100,
+      // color: Colors.amber,
```
Comment (Ali-Alkhars) [inline lib/widgets/timetable_row.dart]: here

File: test/all_saved_timetables_test.dart [hunk 5]
```diff
@@ -0,0 +1,95 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:network_image_mock/network_image_mock.dart';
+import 'package:seg_coursework_app/main.dart';
+import 'package:seg_coursework_app/pages/all_saved_timetables.dart';
+import 'package:seg_coursework_app/pages/visual_timetable_interface.dart';
+import 'package:seg_coursework_app/widgets/picture_grid.dart';
+
+// class TestNavigatorObserver extends NavigatorObserver {
```
Comment (Ali-Alkhars) [inline test/all_saved_timetables_test.dart]: here

**Review by dmunipi1 — APPROVED** [review]
Everything looks good, well written clean code with appropriate testing.

Thank you for your work

**Review by Ali-Alkhars — CHANGES_REQUESTED** [review]
That's some quality work! Thank you.

**Here are some minor stuff to change/add:**
- You should remove commented out code. I have tagged them for you (check the comments below)
- You should add general comments at the top of the pages and widgets files explaining what they do/are used for
- Also comments on top of methods explaining what they do or why they are added

**Comment by skellyhoe** [comment]
> That's some quality work! Thank you.
> 
> **Here are some minor stuff to change/add:**
> 
> * You should remove commented out code. I have tagged them for you (check the comments below)
> * You should add general comments at the top of the pages and widgets files explaining what they do/are used for
> * Also comments on top of methods explaining what they do or why they are added

Thank you for your comments and tags. Implemented all!

**Comment by skellyhoe** [comment]
Just pushed a fix that centers the timetable. Images and arrows now scale depending on screen size. This will make all components of the timetable fit on the screen without scrolling. Hopefully this will be it.

**Review by Ali-Alkhars — APPROVED** [review]
Well done!

**Pull request merged**